src/routes/userRoute.ts

```diff
@@ -1,14 +1,23 @@
 import Router from 'express'
 
 import UserController from '../controllers/UserController'
-import userSchema from '../validations/userValidation'
+import userSchema, { profileValidationalSchema } from '../validations/userValidation'
 import validateRequest from '../utils/validateRequest'
-
+import { singleFileUpload } from '../middlewares/fileUpload'
+import { isOwner } from '../middlewares/authenticationMiddleware'
 const router = Router()
 router.post(
   '/signup',
   validateRequest(userSchema, 'body'),
   UserController.signup,
 )
+router.get(
+  '/profile/:id',isOwner, UserController.getUser)
+
+router.patch(
+  '/profile/:id',isOwner, singleFileUpload,
+  validateRequest(profileValidationalSchema, 'body'),
+  UserController.updateUser
+)
```

Comment:
you can fist pass in isAuthenticated, so that it can decode the authorisation token and pass the decoded in the, 
request
```suggestion
router.get(
  '/profile/:id',isAuthenticated,isOwner, UserController.getUser)

router.patch(
  '/profile/:id',isAuthenticated,isOwner, singleFileUpload,
  validateRequest(profileValidationalSchema, 'body'),
  UserController.updateUser
)
```

Reply:
Here, It seems like here you tried to pass two middleware's which is not necessary , may be you can pass only isAuthenticated middleware, because I have seen that it decodes also the token and also checks if that user is known in our database and then finally set the user into the request.

We don't need to take id in params, we can take all information from the token.

Reply:
Here, It seems like here you tried to pass two middleware's which is not necessary , may be you can pass only isAuthenticated middleware, because I have seen that it decodes also the token and also checks if that user is known in our database and then finally set the user into the request.

We don't need to take id in params, we can take all information from the token.